lambdaclass/mirra_backend
# [GH-917] - Deathmatch mode

> [!Warning]
> Merge alongside https://github.com/lambdaclass/champions_of_mirra/pull/2230
## Motivation

Add a new game mode based on the deathmatch concept. All against all mode where the player with most kills wins the match.

Closes #917

## Summary of changes

- Added a new matchmaking for deathmatch mode
- Added a respawn feature only enabled for deathmatch mode
- Added a new win condition only enabled for deathmatch mode
- Send the gamemode to the client

## How to test it?

- Open the browser client and play a game in the deathmatch mode. If you want, you can print the winner in `apps/arena/lib/arena/game_updater.ex:337` and check that it is correct

## Checklist
